sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -293,6 +293,17 @@ public string CorrelationId
         ///
         public long Offset => _amqpMessage.GetOffset(long.MinValue);
 
+        /// <summary>
+        ///   The replication segment of the event when it was received from the associated Event Hub partition.
+        /// </summary>
+        ///
+        /// <value>
+        ///   This value is read-only and will only be populated for events that have been read from geo replication enabled Event Hubs. The default value
+        ///   when not populated is null.
+        /// </value>
+        ///
+        public string ReplicationSegment => _amqpMessage.GetReplicationSegment(null);
```

Comment:
Should we also add a `LastPartitionReplicationSegment`?

Reply:
What about a `PendingPublishReplicationSegment`?

Reply:
We don't need a pending, as that isn't something we assign from the client.   I don't know that the service is going to add a last to the management runtime metrics - probably not.  I don't see that being helpful, so I don't think we need either.